lib/pbench/test/unit/server/query_apis/commons.py

```diff
@@ -129,6 +136,14 @@ def get_expected_status(self, payload: JSON, header: HeaderTypes) -> int:
             expected_status = HTTPStatus.OK
         return expected_status
 
+    def make_request_call(
+        self, client, url, header: JSON, json: JSON = None, data=None
+    ):
+        assert self.api_method != API_METHOD.GET or json is data is None
+
+        func = client.get if self.api_method == API_METHOD.GET else client.post
```

Comment:
Since you have two tests for `API_METHOD.GET`, this might be better as an `if/else` instead of two expressions:

(NOTE, I put two separate assert statements because that makes slightly better messaging if one condition fails; even though that's probably not a big deal in this context.)

```python
    assert self.api_method in (API_METHOD.GET, API_METHOD.POST)
    if self.api_method == API_METHOD.GET:
        assert json is None
        assert data is None
        func = client.get
    else:
        func = client.post
```